### PR TITLE
External developer friendly changes

### PR DESCRIPTION
**Pull request by iowillhoit**
> [!WARNING]
> Do not merge yet, additional PRs incoming
> TODO: Revert `ew/external-friendly` branch refs

We hard coded our svc-cli-bot username/email in a few spots. Making these workflows not useable to external developers. 
This PR looks up the username/password based on the Github Token being used. Additional changes are made to take advantage of this lookup.

Example of a Release working on an external plugin: https://github.com/iowillhoit/plugin-workflow-test/actions/runs/7748104245
Example of this working on an internal plugin (using secrets inherit) https://github.com/salesforcecli/testPackageRelease/actions/runs/7759564485/job/21163873201#step:8:27

[@W-14809762@](https://gus.my.salesforce.com/apex/ADM_WorkLocator?bugorworknumber=W-14809762)

Other PRs:
https://github.com/oclif/plugin-plugins/pull/784
https://github.com/salesforcecli/plugin-trust/pull/720
https://github.com/salesforcecli/plugin-template-sf-external/pull/110/
https://github.com/forcedotcom/source-deploy-retrieve/pull/1226
https://github.com/salesforcecli/cli/pull/1456

### CODE REVIEW

File: .github/actions/getGithubUserInfo/action.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    value: ${{ steps.user-info.outputs.email }}
+
+runs:
+  using: composite
+  steps:
+    - name: Validate token is passed
+      if: inputs.SVC_CLI_BOT_GITHUB_TOKEN == ''
+      uses: actions/github-script@v3
+      with:
+        script: core.setFailed("You must pass a Github Token with Repo Access as SVC_CLI_BOT_GITHUB_TOKEN")
```
Comment (mshanemc) [inline .github/actions/getGithubUserInfo/action.yml:24]: should we say that it needs write access?

File: .github/workflows/create-github-release.yml [hunk 2]
```diff
@@ -3,7 +3,8 @@ on:
   workflow_call:
     secrets:
       SVC_CLI_BOT_GITHUB_TOKEN:
-        description: a github PAT with repo access
+        description: A Github PAT with repo access.
```
Comment (mshanemc) [inline .github/workflows/create-github-release.yml:6]: write access?

File: README.md [hunk 3]
```diff
@@ -2,6 +2,19 @@
...
+> - Create a new PAT with Repo access
+>   - It is recommended that this is a service account user
+> - Add the PAT as an Actions [Organization secret](https://github.com/organizations/salesforcecli/settings/secrets/actions)
+>   - Set the `Name` to `SVC_CLI_BOT_GITHUB_TOKEN`
+>   - Paste in your new PAT as the `Value`
+>   - Set `Repository Access` to 'Selected Repositories'
+>   - Click the gear icon to select repos that need access to the PAT
+>     - This can be edited later
+>   - Click `Add Secret`
+
```
Comment (mshanemc) [inline README.md:17]: I think for the `create-github-release` action, we also needed to ensure that the bot user had the ability to make commits to main.  Some of the gh branch protection rules (ex: nobody can commit to main) were preventing this from working.

it's not important for our stuff, more of a "help externals use this"

File: .github/actions/gitConfig/action.yml [hunk 4]
```diff
@@ -1,11 +1,20 @@
 name: gitConfig
 description: "Sets git username/email and push behavior"
+
+inputs:
```
[the remark this answers is not in the document]
Comment (iowillhoit) [reply .github/actions/gitConfig/action.yml:4]: Perhaps for us, but using this as a default could cause odd commits for externals if they don't pass in the token.

File: .github/workflows/githubRelease.yml [hunk 5]
```diff
@@ -1,4 +1,15 @@
 name: tag and github release
+#
+#
+#
+#
+# -----------------------------------------------------------------------
+# NOTE: This workflow is deprecated in favor of create-github-release.yml
+# -----------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (iowillhoit) [reply .github/workflows/githubRelease.yml:8]: Sounds good 👍

**Review by mshanemc — APPROVED** [review]

**Pull request merged**